### ISSUE TITLE
Not an issue but a question on privacy with BYOD in school

### ISSUE THREAD

**Issue by michelemangen** [veyon/veyon#795]
Hello,

We work in a "bring your own device" system. So students take their laptops home. I wanted to use Veyon in class but I want to be sure that monitoring student activity after school hours can be disabled. 
Excuse my lack of experience with these kind of programs, and thanks for your help

**Comment by SlrG** [comment]
Hello,

TL;DR: Veyon is at the moment not a viable solution in a BYOD environment.

I'm not the developer, the following are just my personal thoughts on that topic.

For Veyon to work, you would need to install the veyon service on the machines of the students. Will they or their parents allow it? I don't know.

Under normal conditions the service is always active but can be disabled by the systems admin, which with BYOD devices are the students or their parents. If they disable it, it won't work anymore until it is enabled again. Will they do it for you to control them in school? I don't know. - Something might be scripted here, to do this programmatically when the students leave and enter the school network, so it is not impossible however - if the students don't sabotage it.

For Veyon to reliably work, I would recommend wired connections only. WLAN connections will probably fail if there is a normal classroom size number of PCs connected. But for BYOD wired connections are not really desired in most cases.

Veyon communicates based on the IPs of the PCs either configured by you directly or by reading LDAP information you provide. For it to work reliably, you will have to make sure, that each PC always gets the same IP.

In a BYOD environment I imagine only key based authentication will be possible, as the students systems will probably not be joined to your domain, which would create trouble for them using their devices at home. If a possible attacker has access to your key files and gets access to the students home network its possible to control (enable mic or webcam) or just watch what is happening on the students pc. But this is probably an unlikely szenario and if an attacker has home network access he might be able to do the same things anyway, no matter if veyon is installed on the students device.

In the end as a teacher or school admin I personally would never install anything on BYOD devices. If they are rental devices owned by the school the situation might be different, but generally even then I think Veyon is not the right solution at the moment.

cheers

SlrG

**Comment by procc1** [comment]
Very easy. Create a schedule to stop/start Veyon service. For example on Windows systems, you can do this with "Task scheduler".
